ld_script.txt

```diff
@@ -600,7 +600,7 @@ SECTIONS {
         src/debug/unknown_debug_menu.o(.rodata);
         data/strings2.o(.rodata);
         src/engine/menu_cursor.o(.rodata);
-        data/mystery_event_msg.o(.rodata);
+        src/engine/mystery_event_script.o(.rodata);
```

Comment:
This causes the order of files to differ between the .text and .rodata sections. mystery_event_msg is probably its own file, not part of mystery_event_script.

Reply:
How important is preserving file order? There appear to be other instances where file order matching is violated 🤔 💭

Reply:
Well, it's only important if we want this source to resemble the original source code. The other violations should be fixed as well, if possible.